pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -580,4 +615,120 @@ private static IndexingConfig sanitizeIndexingConfig(IndexingConfig indexingConf
+        }
+      } catch (NumberFormatException e) {
+        throw new IllegalStateException("Invalid replication number", e);
+      }
+    }
+
+    if (verifyReplicasPerPartition) {
+      String replicasPerPartitionStr = segmentsConfig.getReplicasPerPartition();
+      if (replicasPerPartitionStr == null) {
+        throw new IllegalStateException("Field replicasPerPartition needs to be specified");
+      }
+      try {
+        int replicasPerPartition = Integer.parseInt(replicasPerPartitionStr);
+        if (replicasPerPartition < defaultTableMinReplicas) {
+          LOGGER.info(
+              "Creating table with minimum replicasPerPartition of: {} instead of requested replicasPerPartition: {}",
+              defaultTableMinReplicas, replicasPerPartition);
+          segmentsConfig.setReplicasPerPartition(String.valueOf(defaultTableMinReplicas));
+        }
+      } catch (NumberFormatException e) {
+        throw new IllegalStateException("Invalid value for replicasPerPartition: '" + replicasPerPartitionStr + "'", e);
+      }
+    }
+  }
+
+  /**
+   * Ensure the table config has storage quota set as per cluster configs.
+   * If it doesn't, set the quota config into the table config
+   */
+  public static void ensureStorageQuotaConstraints(TableConfig tableConfig, String maxAllowedSize) {
+    // Dim tables must adhere to cluster level storage size limits
+    if (tableConfig.isDimTable()) {
+      QuotaConfig quotaConfig = tableConfig.getQuotaConfig();
+      long maxAllowedSizeInBytes = DataSizeUtils.toBytes(maxAllowedSize);
+
+      if (quotaConfig == null) {
+        // set a default storage quota
+        tableConfig.setQuotaConfig(new QuotaConfig(maxAllowedSize, null));
+        LOGGER.info("Assigning default storage quota ({}) for dimension table: {}", maxAllowedSize,
+            tableConfig.getTableName());
+      } else {
+        if (quotaConfig.getStorage() == null) {
+          // set a default storage quota and keep the RPS value
+          tableConfig.setQuotaConfig(new QuotaConfig(maxAllowedSize, quotaConfig.getMaxQueriesPerSecond()));
+          LOGGER.info("Assigning default storage quota ({}) for dimension table: {}", maxAllowedSize,
+              tableConfig.getTableName());
+        } else {
+          if (quotaConfig.getStorageInBytes() > maxAllowedSizeInBytes) {
+            throw new IllegalStateException(String
+                .format("Invalid storage quota: %d, max allowed size: %d", quotaConfig.getStorageInBytes(),
+                    maxAllowedSizeInBytes));
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Consistency checks across the offline and realtime counterparts of a hybrid table
+   */
+  public static void verifyHybridTableConfigs(String rawTableName, TableConfig offlineTableConfig,
+      TableConfig realtimeTableConfig) {
+    if (offlineTableConfig == null || realtimeTableConfig == null) {
+      return;
+    }
```

Comment:
The passed in config should not be null. You can actually add a precondition on them not null to guard the case where the table config fetch failed in `PinotTableRestletResource.checkHybridTableConfig()`